### DISCUSSION TITLE
Delay gameplay until quick retry is released

### DISCUSSION THREAD

**Discussion by goaaats** [ppy/osu#17010] [Ideas]
This discussion was created in response to PR #16988, which implements a way to "delay" gameplay on the loading screen until the quick retry key is released in order to get ready to play, e.g. by adjusting your pen grip or position.

I've used this a lot in stable, as just holding a button between tries to readjust is easier/less distracting than having to pause, and found myself missing the behavior when playing more lazer as of late.

Some comments from the PR:
> Delaying gameplay start is a useful feature. I have used this in stable during normal gameplay to adjust my pen grip, reposition my tablet, etc. In lazer, there are times when the countdown before the first circle is too short and I end up spamming quick retry to buy time.

> It would be a nice optional setting to have for those who are used to or prefer the way stable behaves

> I disagree with "adding a setting" for this even more than I disagree with the original proposal. This should be an opinionated choice. As in either the UX should either stay as is or be changed to the proposed one in the OP, but having toggles for every single little behaviour is terrible and does not scale.

> I'm confused. I think pause should be used rather than spamming retry for getting time to adjust input.

> It is much easier to just hold down the same key you're already pressing than to pause, having to press "continue" again and deal with the "click the orange cursor to resume" thing

**Comment by LittleEndu** [comment]
> I'm confused. I think pause should be used rather than spamming retry for getting time to adjust input.

There are people who play with their retry key close to their game play buttons while their pause key is several rows away. "Just pause instead" is not a solution for those players. And these players wouldn't want to spam the retry key (which would always be possible, would just retry as soon as gameplay starts), they want to hold it (which right now they can't). When you want to readjust your grip or move mouse wire or what ever, you don't want to split your attention.

**Comment by Walavouchey** [comment]
Opening any overlay or hovering over the visual settings when starting a beatmap also delays gameplay, so in a sense there are already a few keybinds available for this purpose.

**Comment by bdach** [comment]
I would also like to say that even if this is brought back I don't like the way it is currently implemented in the aforementioned PR. Why would pressing the key down delay at the loader level rather than keeping the gameplay screen dimmed until the user releases?

**Reply by peppy** [reply]
For sure, changing the transition to only complete when the key is released should be the only way this is implemented if it's decided that it's what we want.

I believe I intentionally made it work this way in stable so it's probably fine to do the same here. Although I also intentionally changed it in lazer as the new behaviour felt better to me, but if it's something relied upon I'm not going to question reverting to the stable way.
